app/src/main/java/org/httpsknighthacks/knighthacksandroid/MainActivity.java

```diff
@@ -67,31 +71,29 @@ protected void onCreate(Bundle savedInstanceState) {
             manager.createNotificationChannel(channel);
         }
 
+        subscribeToTopic();
+
+        // End of firebase code.
+
         setContentView(R.layout.activity_main);
 
         mRequestQueueSingleton = new RequestQueueSingleton(getApplicationContext());
         getImageAndTitles();
+    }
 
-        // Shows the Token to the screen
-
-        /*FirebaseInstanceId.getInstance().getInstanceId()
-                .addOnCompleteListener(new OnCompleteListener<InstanceIdResult>() {
+    private void subscribeToTopic() {
+        FirebaseMessaging.getInstance().subscribeToTopic("general")
+                .addOnCompleteListener(new OnCompleteListener<Void>() {
                     @Override
-                    public void onComplete(@NonNull Task<InstanceIdResult> task) {
+                    public void onComplete(@NonNull Task<Void> task) {
+                        String msg = "Successful: Subscribed to General";
                         if (!task.isSuccessful()) {
-                            Log.w("MainActivity", "getInstanceId failed", task.getException());
-                            return;
+                            msg = "Failed";
                         }
-
-                        // Get new Instance ID token
-                        String token = task.getResult().getToken();
-
-                        // Log and toast
-                        String msg = getString(R.string.msg_token_fmt, token);
-                        Log.d("MainActivity", msg);
+                        Log.d("Subscription to general", msg);
```

Comment:
Reuse of `general`,
Move to single constant would probably be best, in case we need to change the value or something. It makes it easier to do that change later on because it will only need to be made in one place.